src/transformers/generation/utils.py

```diff
@@ -4290,6 +4290,7 @@ def _assisted_decoding(
                     dim=0,
                 )
 
+            candidate_input_ids = candidate_input_ids.to(self.device)
```

Comment:
in a previous PR https://github.com/keyboardAnt/transformers/pull/4/files I proposed to do it just after `get_candidates` at the same place we move `candidate_logits` to `self.device`

Reply:
oke, might be better to move it there for easier readability :)